vault/audit.go

```diff
@@ -617,3 +628,29 @@ func (g genericAuditor) AuditResponse(ctx context.Context, input *logical.LogInp
 	logInput.Type = g.mountType + "-response"
 	return g.c.auditBroker.LogResponse(ctx, &logInput)
 }
+
+// hasInvalidAuditOptions is used to determine if a non-Enterprise version of Vault
+// is being used when supplying options that contain options exclusive to Enterprise.
+func hasInvalidAuditOptions(options map[string]string) bool {
```

Comment:
This is way out of scope for this PR, but I thought I'd capture it in a comment anyway: given the discussion we'd had previously about it being possible to define 'invalid' audit options in previous versions of Vault, since we'll be adding new features in the future, it might make sense to prevent that behaviour for any invalid version